mithril-signer/src/runtime/state_machine.rs

```diff
@@ -181,74 +184,80 @@ impl StateMachine {
+                    pending_certificate: &CertificatePending,
+                ) -> bool {
+                    Some(&pending_certificate.signed_entity_type)
+                        == last_signed_entity_type.as_ref()
+                }
+
                 if let Some(new_epoch) = self.has_epoch_changed(*epoch).await? {
-                    info!(" → new Epoch detected, transiting to UNREGISTERED");
+                    info!("→ Epoch has changed, transiting to UNREGISTERED");
                     *state = self
-                        .transition_from_signed_to_unregistered(new_epoch)
+                        .transition_from_ready_to_sign_to_unregistered(new_epoch)
                         .await?;
-                } else if let Some(pending_certificate) =
-                    self.runner.get_pending_certificate().await.map_err(|e| {
+                } else {
+                    match self.runner.get_pending_certificate().await.map_err(|e| {
                         RuntimeError::KeepState {
                             message: "could not fetch the pending certificate".to_string(),
                             nested_error: Some(e),
                         }
-                    })?
-                {
-                    info!(
-                        " ⋅ Epoch has NOT changed but there is a pending certificate";
-                        "pending_certificate" => ?pending_certificate
-                    );
-                    if pending_certificate.signed_entity_type == *signed_entity_type {
-                        info!(" ⋅ pending certificate has not changed, waiting…");
-                    } else {
-                        info!(" → new pending certificate detected, transiting to REGISTERED");
-                        *state = self.transition_from_signed_to_registered(*epoch).await?;
+                    })? {
+                        Some(pending_certificate)
+                            if is_same_signed_entity_type(
+                                last_signed_entity_type,
+                                &pending_certificate,
+                            ) =>
+                        {
+                            info!(" ⋅ same entity type, cannot sign pending certificate, waiting…");
+                        }
+                        Some(pending_certificate)
+                            if self
+                                .runner
+                                .can_sign_signed_entity_type(
+                                    &pending_certificate.signed_entity_type,
+                                )
+                                .await =>
+                        {
+                            info!(
+                                " ⋅ Epoch has NOT changed but there is a pending certificate";
+                                "pending_certificate" => ?pending_certificate
+                            );
+                            info!(" → we can sign this certificate, transiting to READY_TO_SIGN");
```

Comment:
If we do that, in log file we'll have only one line:
```
{"msg":" ⋅ Epoch has NOT changed but there is a pending certificate → we can sign this certificate, transiting to READY_TO_SIGN"
```
instead of two lines
```
{"msg":" ⋅ Epoch has NOT changed but there is a pending certificate"
{"msg":" → we can sign this certificate, transiting to READY_TO_SIGN"
```
Is it ok for you ?